config/routes.rb

```diff
@@ -3,5 +3,17 @@
   devise_for :users
   resources :products
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  
+  # match "/*path" => redirect("/?goto=%{path}")
+
+  class XHRConstraint
```

Comment:
We will eventually want to refactor this class out or routes and declare it somewhere else in the load path. That being said, if this works locally then KICK ASS job Sean! We can deal with style guidesque shit in future iteration.